src/energyplus/Test/EMS_GTest.cpp

```diff
@@ -2230,32 +2345,80 @@ TEST_F(EnergyPlusFixture, ForwardTranslatorActuator_API3_EMS) {
-  OptionalString name0 = objects[0].getString(EnergyManagementSystem_ActuatorFields::ActuatedComponentUniqueName);
-  OptionalString name1 = objects[1].getString(EnergyManagementSystem_ActuatorFields::ActuatedComponentUniqueName);
-  std::string test1 = "Thermal Zone 1 Lights 1";
-  std::string test2 = "Thermal Zone 2 Lights 1";
-  std::string test3 = "Thermal Zone 3 Lights 1";
-  std::string test4 = "Thermal Zone 4 Lights 1";
-  //ONLY 1 TZ in model now so should default back to that
-  EXPECT_EQ(name0.get(), test1);
-  EXPECT_EQ(name1.get(), test1);
-  //EXPECT_TRUE((name0.get() == test1) || (name0.get() == test2) || (name0.get() == test3) || (name0.get() == test4));
-  //EXPECT_TRUE((name1.get() == test1) || (name1.get() == test2) || (name1.get() == test3) || (name1.get() == test4));
-  //model.save(toPath("./EMS_actuator_exampleModel2_electric.osm"), true);
-  //workspace.save(toPath("./EMS_actuator_exampleModel2_electric.idf"), true);
+  ForwardTranslator ft;
+
+  // When excluding space translation (historical behavior)
+  {
+    ft.setExcludeSpaceTranslation(true);
+
+    Workspace workspace = ft.translateModel(model);
+
+    // We get only a warning, since we can recover from it easily
+    EXPECT_EQ(0u, ft.errors().size());
+    EXPECT_EQ(2, ft.warnings().size());
+    EXPECT_EQ(
+      "Actuator 'Energy_Management_System_Actuator_1' references a SpaceLoad 'Lights 1' attached to the SpaceType 'Space Type 1' but you have turned "
+      "off ForwardTranslation's Space Feature. Falling back to using the attached ThermalZone 'Thermal Zone 1' since there is only 1.",
+      ft.warnings().front().logMessage());
+    EXPECT_EQ(
+      "Actuator 'Energy_Management_System_Actuator_2' references a SpaceLoad 'Lights 1' attached to the SpaceType 'Space Type 1' but you have turned "
+      "off ForwardTranslation's Space Feature. Falling back to using the attached ThermalZone 'Thermal Zone 1' since there is only 1.",
+      ft.warnings().back().logMessage());
+
+    std::vector<WorkspaceObject> objects = workspace.getObjectsByType(IddObjectType::EnergyManagementSystem_Actuator);
+    //expect 2 actuators
+    //ACTUATORS WILL STILL GET TRANSLATED WITH BLANK ZONENAME FIELD
+    ASSERT_EQ(2, objects.size());
+
+    std::vector<std::string> actuatedNames;
+    std::transform(objects.cbegin(), objects.cend(), std::back_inserter(actuatedNames),
+                   [](const auto& wo) { return wo.getString(EnergyManagementSystem_ActuatorFields::ActuatedComponentUniqueName).get(); });
+    std::sort(actuatedNames.begin(), actuatedNames.end());
+
+    // TODO: hum we end up with two actuators on the same zone, this is problematic... We did issue helpful warnings, and let E+ catch the issue.
+    // Can't always protect the user against themselves
+    EXPECT_EQ("Thermal Zone 1 Lights 1", actuatedNames.at(0));
+    EXPECT_EQ("Thermal Zone 1 Lights 1", actuatedNames.at(1));
```

Comment:
Note this case...

Reply:
its a tricky one, but i agree we've done all we can reasonably do